tests/base/array_view_15.cc

```diff
@@ -0,0 +1,54 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// test for class ArrayView for scalar type
+
+#include <deal.II/base/array_view.h>
+
+#include "../tests.h"
+
+
+void
+test()
+{
+  int v = 9;
+
+  ArrayView<int> a1(v); // writable view
+
+  Assert(a1.size() == 1, ExcInternalError());
```

Comment:
I suggest we use `AssertThrow` to ensure correctness even in the release mode.

Reply:
@rezarastak `AssertThrow` is reserved for conditions that you would normally handle with the C++ exception mechanism. Typically, when an error condition depends on user input or configuration (for example throwing `SolverControl::NoConvergence`). Here, however, an invalid array view is an internal error that should be handled by `Assert`. It is our choice that we remove `Assert`s in release mode.

Reply:
Thank you for explaining @tamiko. I think I understand the overall idea and the difference between `Assert` and `AssertThrow` in the code of the library. I want to know what is the policy of dealii developers regarding assert statements in the tests. My opinion is that a test should assert all reasonable truth statements (to ensure correctness) in both debug and release modes. We are not concerned with the cpu efficiency of the code written for test and we are not worried about the bloated code size of the test binary either. So we should make sure that during compilation of tests, the `Assert` statements are not removed in the release mode. Maybe the build system already does this and I am not aware. Alternatively, we should replace all `Assert` s with `AssertThrow` in the test codes. I already know many dealii tests that use `AssertThrow` frequently.

Reply:
I don't think we have a policy for tests, but I seem to remember that we converted some `Assert` statements in tests to `AssertThrow` for exactly the reason you mention.

I would be curious to see what happens if one converted all of them. Want to give that a try? But let's leave this for a separate patch :-)

Reply:
Sure! I'll provide a patch regarding the use of `Assert` in tests.